runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -41,11 +44,27 @@ public StreamViewSMRAdapter(CorfuRuntime runtime,
         this.streamView = streamView;
     }
 
+    private List<SMREntry> dataAndCheckpointMapper(ILogData logData) {
+        if (logData.hasCheckpointMetadata()) {
+            // This is a CHECKPOINT record.  Extract the SMREntries, if any.
+            CheckpointEntry cp = (CheckpointEntry) logData.getPayload(runtime);
```

Comment:
!@#$! GH notifications aren't being delivered to me, sorry about the delay today.  I've moved ref fixup to `StreamViewSMRAdapter::dataAndCheckpointMapper` in commit d1d019e.